packages/core/core/src/services/document-service/transform/__tests__/filters.test.ts

```diff
@@ -0,0 +1,136 @@
+import { LoadedStrapi } from '@strapi/types';
+import { PRODUCT_UID, CATEGORY_UID, models } from './utils';
+
+import { transformFilters } from '../filters';
+
+const findProducts = jest.fn(() => ({}));
+const findCategories = jest.fn(() => ({}));
+
+const findManyQueries = {
+  [PRODUCT_UID]: findProducts,
+  [CATEGORY_UID]: findCategories,
+} as Record<string, jest.Mock>;
+
+describe('transformFilters', () => {
+  beforeAll(() => {
+    global.strapi = {
+      getModel: (uid: string) => models[uid],
+      db: {
+        query: jest.fn((uid) => ({ findMany: findManyQueries[uid] })),
+        metadata: {
+          get: jest.fn(() => ({
+            columnToAttribute: [],
+          })),
+        },
+      },
+    } as unknown as LoadedStrapi;
+  });
+
+  it('should transform simple filters', async () => {
+    const input = { id: 'someValue' };
+    const expected = { documentId: 'someValue' };
+
+    expect(
+      await transformFilters(input, {
+        uid: CATEGORY_UID,
+      })
+    ).toEqual(expected);
+  });
+
+  it('should handle empty objects', async () => {
+    const input = {};
+    const expected = {};
+
+    expect(await transformFilters(input, { uid: PRODUCT_UID })).toEqual(expected);
+  });
+
+  it('should not modify other fields that are in the model schema', async () => {
+    const input = { name: 'value', id: 'test' };
+    const expected = { name: 'value', documentId: 'test' };
+
+    expect(await transformFilters(input, { uid: PRODUCT_UID })).toEqual(expected);
+  });
+
+  it('should not modify other fields not in the model schema', async () => {
+    const input = { otherField: 'value', id: 'test' };
+    const expected = { otherField: 'value', documentId: 'test' };
+
+    expect(await transformFilters(input, { uid: PRODUCT_UID })).toEqual(expected);
+  });
+
+  it('should ignore non relational nested filters', async () => {
+    const input = { _tmp: { id: 'nestedValue' } };
+
+    expect(await transformFilters(input, { uid: PRODUCT_UID })).toEqual(input);
+  });
+
+  it('should ignore non relational nested array filters', async () => {
+    const input = { _tmp: [{ id: 'arrayValue1' }, { id: 'arrayValue2' }] };
+
+    expect(await transformFilters(input, { uid: PRODUCT_UID })).toEqual(input);
+  });
+
+  it('should handle nested relational filters', async () => {
+    const input = { category: { id: 'nestedValue' } };
+    const expected = { category: { documentId: 'nestedValue' } };
+
+    expect(await transformFilters(input, { uid: PRODUCT_UID })).toEqual(expected);
+  });
+
+  it('should handle arrays in relational filters', async () => {
+    const input = { categories: [{ id: 'arrayValue1' }, { id: 'arrayValue2' }] };
+    const expected = {
+      categories: [{ documentId: 'arrayValue1' }, { documentId: 'arrayValue2' }],
+    };
```

Comment:
could you are tests for more complex filters, like:

const input = { categories: [{ id: { $eq: 'arrayValue1' }}}] };

This can go even deeper if I recall correctly: 

const input = { categories: [{ id: { $and: [{$eq: 'arrayValue1' },{$eq: 'arrayValue1' } ]}}] };

Reply:
Just realized this transformation should be a little bit more complex,

shouldn't we specify the documentId + the locale and publication state of the filtering?
```
    const input = { categories: [{ id: 'arrayValue1' }}] };
    const expected = {
      categories: [{ documentId: 'arrayValue1' , locale: 'en', publishedAt: null}],
    };
```

Specially weh filtering relations I believe this might be necessary 🤔